### ISSUE TITLE
Add support for differential push

### ISSUE THREAD

**Issue by wouterdb** [inmanta/inmanta#246]
When the agent request the model, all resources are sent by the server.

It would be good to send only changes wrt the previous version.

depends on #245

**Comment by bartv** [comment]
This would allow us to use a hybrid model:
For push: only send the diff
For pull from the agent: send the full model

**Comment by bartv** [comment]
Duplicate of #47
